libs/geo/src/main/java/org/opensearch/geometry/utils/WellKnownText.java

```diff
@@ -294,23 +307,72 @@ private Geometry parseGeometry(StreamTokenizer stream) throws IOException, Parse
             case "bbox":
                 return parseBBox(stream);
             case "geometrycollection":
-                return parseGeometryCollection(stream);
+                throw new IllegalStateException("Unexpected type: geometrycollection");
             case "circle": // Not part of the standard, but we need it for internal serialization
                 return parseCircle(stream);
         }
         throw new IllegalArgumentException("Unknown geometry type: " + type);
     }
 
+    /**
+     * Parse geometry collection iteratively
+     *
+     * Parsing geometry collection recursively can lead to StackOverflowError when there is a deeply nested structure of GeometryCollection
```

Comment:
I don't think we need to add comment why/not we are doing this recursively or iteratively. 
What can help here is a brief javadoc on what and how does this method works.

Reply:
How about remove the comment completely?
For what and how, I think it is all there in the method name and code itself. Otherwise, please suggest one.

Reply:
Yeah, this makes sense.
Mentioning recursively/iteratively did not added any value and perhaps would have created confusion later.